core/tests/test_views.py

```diff
@@ -72,39 +94,66 @@ def test_add_post_unauthorized(client: Client, community: Community) -> None:
 
 
 def test_add_comment_valid(client: Client, user: User, post: Post) -> None:
-    data = {
-        "parent_id": post.pk,
-        "content": "This is a test comment content.",
-    }
-    client.force_login(user)
-    assert post.children_count == 0
-    assert post.get_comments().count() == 0
-    response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data, follow=True)
-    assert response.status_code == 200
-    assert post.children_count == 1
-    assert post.get_comments().count() == 1
-    post.refresh_from_db()
-    assert response.context["comments"][0].author == user
-    assert response.context["comments"][0].content == data["content"]
+    with tempfile.NamedTemporaryFile(suffix=".jpg", delete=False) as tmp_file:
+        image = Image.new("RGB", (100, 100), color=(73, 109, 137))
+        image.save(tmp_file, format="JPEG")
+        tmp_file.seek(0)
+
+        avatar_file = SimpleUploadedFile(name="test_avatar.jpg", content=tmp_file.read(), content_type="image/jpeg")
+
+        user.avatar = avatar_file
+        user.save()
+        data = {
+            "parent_id": post.pk,
+            "content": "This is a test comment content.",
+        }
+        client.force_login(user)
+        assert post.children_count == 0
+        assert post.get_comments().count() == 0
+        response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data, follow=True)
+        assert response.status_code == 200
+        assert post.children_count == 1
+        assert post.get_comments().count() == 1
+        post.refresh_from_db()
+        assert response.context["comments"][0].author == user
+        assert response.context["comments"][0].content == data["content"]
 
 
 def test_add_comment_invalid(client: Client, user: User, post: Post) -> None:
-    data = {"parent_id": post.pk, "content": ""}
-    client.force_login(user)
-    response = client.post(reverse("post-detail", kwargs={"pk": post.pk}), data=data)
-    assert response.status_code == 200
-    form = response.context["form"]
-    assert len(form.errors) == 1
-    assert "This field is required." in form.errors["content"]
+    with tempfile.NamedTemporaryFile(suffix=".jpg", delete=False) as tmp_file:
```

Comment:
Duplicated code. We don't need it on testing validation, like this invalid tests. We do need on create though. The reason is on create the result shows avatar, on error it does not - there is no post/comment to display the avatar on.

Reply:
I fixed it.